pymc_marketing/clv/models/pareto_nbd.py

```diff
@@ -144,63 +152,107 @@ class ParetoNBDModel(CLVModel):
+           "Counting Your Customers: Who Are They and What Will They Do Next".
            Management Science,Vol. 33, No. 1 (Jan., 1987), pp. 1-24.
     .. [2] Fader, Peter & G. S. Hardie, Bruce (2005).
-           "A Note on Deriving the Pareto/NBD Model and Related Expressions."
+           "A Note on Deriving the Pareto/NBD Model and Related Expressions".
            http://brucehardie.com/notes/009/pareto_nbd_derivations_2005-11-05.pdf
     .. [3] Fader, Peter & G. S. Hardie, Bruce (2014).
-           "Additional Results for the Pareto/NBD Model."
+           "Additional Results for the Pareto/NBD Model".
            https://www.brucehardie.com/notes/015/additional_pareto_nbd_results.pdf
     .. [4] Fader, Peter & G. S. Hardie, Bruce (2014).
-           "Deriving the Conditional PMF of the Pareto/NBD Model."
+           "Deriving the Conditional PMF of the Pareto/NBD Model".
            https://www.brucehardie.com/notes/028/pareto_nbd_conditional_pmf.pdf
+    .. [5] Fader, Peter & G. S. Hardie, Bruce (2007).
+           "Incorporating Time-Invariant Covariates into the Pareto/NBD and BG/NBD Models".
+           https://www.brucehardie.com/notes/019/time_invariant_covariates.pdf
     """
 
     _model_type = "Pareto/NBD"  # Pareto Negative-Binomial Distribution
-    _params = ["r", "alpha", "s", "beta"]
 
     def __init__(
         self,
         data: pd.DataFrame,
+        pr_covar_columns: Optional[List[str]] = None,
+        dr_covar_columns: Optional[List[str]] = None,
         model_config: Optional[Dict] = None,
         sampler_config: Optional[Dict] = None,
     ):
-        # Assign inputs to attributes and perform validation checks
-        try:
-            self.customer_id = data["customer_id"]
-            if len(np.unique(self.customer_id)) != len(self.customer_id):
-                raise ValueError("Customers must have unique ID labels.")
-            self.coords = {"customer_id": self.customer_id}
-        except KeyError:
-            raise KeyError("customer_id column is missing from data")
-        try:
-            self.frequency = data["frequency"]
-        except KeyError:
-            raise KeyError("frequency column is missing from data")
-        try:
-            self.recency = data["recency"]
-        except KeyError:
-            raise KeyError("recency column is missing from data")
-        try:
-            self.T = data["T"]
-        except KeyError:
-            raise KeyError("T column is missing from data")
+        # TODO: Write a separate, more comprehensive RFM validation method in a future PR.
+        # Perform validation checks on column names and customer identifiers
+        self._validate_column_names(data, ["customer_id", "frequency", "recency", "T"])
+        if len(np.unique(data["customer_id"])) != len(data["customer_id"]):
+            raise ValueError("Customers must have unique ID labels.")
+
+        # validate data contains specified column names for covariates
+        for _ in zip(
```

Comment:
Give a readable name to the `_` variable. Seems like zip is no longer needed either?